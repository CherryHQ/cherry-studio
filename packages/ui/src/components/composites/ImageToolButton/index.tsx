// Original path: src/renderer/src/components/Preview/ImageToolButton.tsx
import { memo } from 'react'

import { Button } from '../../primitives/button'
import { Tooltip } from '../../primitives/tooltip'

interface ImageToolButtonProps {
  tooltip: string
  icon: React.ReactNode
  onPress: () => void
}

const ImageToolButton = ({ tooltip, icon, onPress }: ImageToolButtonProps) => {
  return (
    <Tooltip content={tooltip} delay={500} closeDelay={0}>
      <Button size="icon" className="rounded-full" onClick={onPress} aria-label={tooltip}>
        {icon}
      </Button>
    </Tooltip>
  )
}

export default memo(ImageToolButton)
