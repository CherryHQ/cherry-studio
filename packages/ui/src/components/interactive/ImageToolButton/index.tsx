// Original path: src/renderer/src/components/Preview/ImageToolButton.tsx
import { Button } from 'antd'
import { memo } from 'react'

import Tooltip from '../../base/Tooltip'

interface ImageToolButtonProps {
  tooltip: string
  icon: React.ReactNode
  onClick: () => void
}

const ImageToolButton = ({ tooltip, icon, onClick }: ImageToolButtonProps) => {
  return (
    <Tooltip placement="top" title={tooltip}>
      <Button shape="circle" icon={icon} onClick={onClick} role="button" aria-label={tooltip} />
    </Tooltip>
  )
}

export default memo(ImageToolButton)
