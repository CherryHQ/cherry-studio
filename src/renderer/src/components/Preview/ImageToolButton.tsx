import { Button, Tooltip } from '@cherrystudio/ui'
import { memo } from 'react'

interface ImageToolButtonProps {
  tooltip: string
  icon: React.ReactNode
  onPress: () => void
}

const ImageToolButton = ({ tooltip, icon, onPress }: ImageToolButtonProps) => {
  return (
    <Tooltip content={tooltip} delay={500} closeDelay={0}>
      <Button radius="full" startContent={icon} onPress={onPress} isIconOnly aria-label={tooltip} />
    </Tooltip>
  )
}

export default memo(ImageToolButton)
