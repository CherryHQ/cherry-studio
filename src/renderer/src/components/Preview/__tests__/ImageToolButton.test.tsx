import { render } from '@testing-library/react'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import ImageToolButton from '../ImageToolButton'

// Mock components
vi.mock('antd', () => ({
  Button: vi.fn(({ children, onClick, ...props }) => (
    <button type="button" data-testid="custom-button" onClick={onClick} {...props}>
      {children}
    </button>
  ))
}))

vi.mock('@cherrystudio/ui', () => ({
  Button: ({ children, onPress, disabled, isDisabled, startContent, ...props }: any) => (
    <button type="button" data-testid="button" onClick={onPress} disabled={disabled || isDisabled} {...props}>
      {startContent}
      {children}
    </button>
  ),
  Tooltip: ({ children }: { children: React.ReactNode }) => <>{children}</>
}))

describe('ImageToolButton', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  const defaultProps = {
    tooltip: 'Test tooltip',
    icon: <span data-testid="test-icon">Icon</span>,
    onClick: vi.fn()
  }

  it('should match snapshot', () => {
    const { asFragment } = render(<ImageToolButton {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
