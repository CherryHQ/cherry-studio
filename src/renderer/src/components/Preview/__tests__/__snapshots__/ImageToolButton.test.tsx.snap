// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageToolButton > should match snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Test tooltip"
    data-testid="button"
    radius="full"
    type="button"
  >
    <span
      data-testid="test-icon"
    >
      Icon
    </span>
  </button>
</DocumentFragment>
`;
